sdk/storage/blobs/inc/az_storage_blobs.h

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_STORAGE_BLOBS_H
+#define AZ_STORAGE_BLOBS_H
+
+#include <az_contract.h>
+#include <az_http_pipeline.h>
+#include <az_http_response.h>
+#include <az_result.h>
+#include <az_span.h>
+#include <az_str.h>
+
+#include <stddef.h>
+
+#include <_az_cfg_prefix.h>
+
+
+AZ_NODISCARD AZ_INLINE az_span az_storage_blobs_blob_header_blob_type() { return AZ_STR("x-ms-blob-type"); }
+AZ_NODISCARD AZ_INLINE az_span az_storage_blobs_blob_header_Content_Length() { return AZ_STR("Content_Length"); }
+
+AZ_NODISCARD AZ_INLINE az_span az_storage_blobs_blob_type_AppendBlob() { return AZ_STR("AppendBlob");} /* UnSupported */
+AZ_NODISCARD AZ_INLINE az_span az_storage_blobs_blob_type_BlockBlob() { return AZ_STR("BlockBlob");}
+AZ_NODISCARD AZ_INLINE az_span az_storage_blobs_blob_type_PageBlob() { return AZ_STR("PageBlob"); } /* UnSupported */
+
+typedef struct {
+  az_span service_version;
+  az_http_policy_retry_options retry;
+} az_storage_blobs_blob_client_options;
+
+typedef struct {
+  az_span option;
+} az_storage_blobs_blob_upload_options;
+
+typedef struct {
+  az_span option;
+} az_storage_blobs_blob_download_options;
+
+typedef struct {
+  az_span uri;
+  az_span blob_type;
+  az_http_pipeline pipeline;
+  az_storage_blobs_blob_client_options client_options;
+} az_storage_blobs_blob_client;
+
+extern az_storage_blobs_blob_client_options const AZ_STORAGE_BLOBS_BLOB_CLIENT_DEFAULT_OPTIONS;
+
+/**
+ * @brief Init a client with default options
+ * This is convinient method to create a client with basic settings
+ * Options can be updated specifally after this for unique customization
+ *
+ * Use this, for instance, when only caring about setting one option by calling this method and then
+ * overriding that specific option
+ */
+AZ_NODISCARD AZ_INLINE az_result
+az_storage_blobs_blob_client_options_init(az_storage_blobs_blob_client_options * const options) {
+  AZ_CONTRACT_ARG_NOT_NULL(options);
+  *options = AZ_STORAGE_BLOBS_BLOB_CLIENT_DEFAULT_OPTIONS;
+  return AZ_OK;
+}
+
+AZ_NODISCARD AZ_INLINE az_result az_storage_blobs_blob_client_init(
+    az_storage_blobs_blob_client * const client,
+    az_span const uri,
+    void * const credential,
+    az_storage_blobs_blob_client_options const * const options) {
+  AZ_CONTRACT_ARG_NOT_NULL(client);
+
+  client->uri = uri;
+  // use default options if options is null. Or use customer provided one
+  if (options == NULL) {
+    client->client_options = AZ_STORAGE_BLOBS_BLOB_CLIENT_DEFAULT_OPTIONS;
+  } else {
+    client->client_options = *options;
+  }
+
+  client->pipeline = (az_http_pipeline){
+    .policies = {
+      { .pfnc_process = az_http_pipeline_policy_uniquerequestid, .data = NULL },
```

Comment:
nit: IMHO, we should think how we not to copy paste this code from one client to another. Especially if the function and data is disconnected it increases the probability of errors. For example, if one of the policies starts to accept some data instead of `NULL` we must remember to change it in all clients for all services and compiler will not help us.

Reply:
Not all clients will use the same pipeline of policies. So, we do copy/paste now in all languages but then tweak for specific services.

Reply:
I agree it would be nice to have a way to construct them centrally.  however as Jeff points out, not all clients will use all policies, and not all clients will configure them the same.

Reply:
IMHO, we, at least, should connect data and a function. Something like this:
```c
client->pipeline = (az_http_pipeline){ 
    .policies = { 
      az_http_pipeline_policy_uniquerequest_create(….),
````